activerecord/lib/active_record/integration.rb

```diff
@@ -52,25 +61,47 @@ def to_param
     # used to generate the key:
     #
     #   Person.find(5).cache_key(:updated_at, :last_reviewed_at)
+    #
+    # If ActiveRecord::Base.cache_versioning is turned on, no version will be included
+    # in the cache key. The version will instead be supplied by #cache_version. This
+    # separation enables recycling of cache keys.
+    #
+    #   Product.cache_versioning = true
+    #   Product.new.cache_key     # => "products/new"
+    #   Person.find(5).cache_key  # => "people/5" (even if updated_at available)
     def cache_key(*timestamp_names)
       if new_record?
         "#{model_name.cache_key}/new"
       else
-        timestamp = if timestamp_names.any?
-          max_updated_column_timestamp(timestamp_names)
+        if cache_version
```

Comment:
Got it. From this usecase: when `cache_key` is called with argument, the `cache_version` setting should be ignored.